graylog2-server/src/main/java/org/graylog/failure/DefaultFailureHandler.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.failure;
+
+import com.google.common.collect.ImmutableMap;
+import org.graylog2.indexer.IndexFailureImpl;
+import org.graylog2.indexer.IndexFailureService;
+
+import javax.inject.Inject;
+
+/**
+ * A fallback failure handler, which persists submitted failures in Mongo via {@link IndexFailureService}.
+ * Only indexing failures supported.
+ */
+public class DefaultFailureHandler implements FailureHandler {
+
+    private final IndexFailureService indexFailureService;
+
+    @Inject
+    public DefaultFailureHandler(IndexFailureService indexFailureService) {
+        this.indexFailureService = indexFailureService;
+    }
+
+    @Override
+    public void handle(FailureBatch failureBatch) {
+        failureBatch.getFailures().forEach(failure ->
+                indexFailureService.saveWithoutValidation(new IndexFailureImpl(ImmutableMap.<String, Object>builder()
+                    .put("letter_id", failure.failedMessage().getMessageId())
```

Comment:
This will change the `Letter ID` from UUID to ULID. As @mpfz0r pointed out, the UUID might be helpful to find related information in the server log when users bump their log levels. So we should probably keep the old behaviour.

![image](https://user-images.githubusercontent.com/886541/130468975-e1ef4819-0b65-4d52-8288-28efc64b3d56.png)